triedb/hashdb/database.go

```diff
@@ -528,8 +530,16 @@ func (c *cleaner) Put(key []byte, rlp []byte) error {
 	return nil
 }
 
+// Delete removes the referenced node from the clean cache.
+// This is called during the replay phase of a commit operation
+// where the trie node has been deleted from the database.
 func (c *cleaner) Delete(key []byte) error {
```

Comment:
The delete function of cleaner is never used. Also, hashdb will be deprecated and we don't plan to introduce any unnecessary changes to it.